onnxruntime/test/providers/kernel_def_hash_test.cc

```diff
@@ -156,7 +160,16 @@ TEST(KernelDefHashTest, DISABLED_PrintCpuKernelDefHashes) {
 }
 
 TEST(KernelDefHashTest, ExpectedCpuKernelDefHashes) {
+#if !defined(DISABLE_OPTIONAL_TYPE)
   const bool is_strict = ParseEnvironmentVariableWithDefault<bool>(kStrictKernelDefHashCheckEnvVar, false);
+#else
+  // In builds with the optional type disabled, we cannot do a strict checking of kernel def hashes
```

Comment:
I think there's only one place using strict checking:
https://github.com/microsoft/onnxruntime/blob/5ad6dbb314f3a2dd10450273ceeeb0513cad79b3/tools/ci_build/github/linux/ort_minimal/build_full_ort_and_create_ort_files.sh#L35

if that still works, could we not ignore the environment variable? then it would be obvious if the case that we care about breaks.